portal-bridge/src/bridge/state.rs

```diff
@@ -131,27 +149,149 @@ impl StateBridge {
+                                code,
+                            )
+                            .await?;
+
+                            // gossip contract storage
+                            let storage_changed_nodes =
+                                state.database.get_storage_trie_diff(address);
+
+                            let storage_walk_diff =
+                                TrieWalker::new(account.storage_root, storage_changed_nodes);
+
+                            for storage_node in storage_walk_diff.nodes.keys() {
+                                let storage_proof = storage_walk_diff.get_proof(*storage_node);
+                                self.gossip_storage(
+                                    &account_proof,
+                                    &storage_proof,
+                                    address,
+                                    block_tuple.header.header.hash(),
+                                )
+                                .await?;
+                            }
+                        }
+                    }
+                }
             }
+            // flush the database cache
+            // This is used for gossiping storage trie diffs
+            state.database.storage_cache.clear();
         }
         Ok(())
     }
 
+    async fn gossip_account(
+        &self,
+        account_proof: &AccountProof,
+        block_hash: B256,
+    ) -> anyhow::Result<()> {
+        let permit = self
+            .gossip_limit_semaphore
+            .clone()
+            .acquire_owned()
+            .await
+            .expect("to be able to acquire semaphore");
+        let content_key = create_account_content_key(account_proof)?;
+        let content_value = create_account_content_value(block_hash, account_proof)?;
+        Self::spawn_serve_state_proof(
+            self.portal_client.clone(),
+            content_key,
+            content_value,
+            permit,
+            self.metrics.clone(),
+        );
+        Ok(())
+    }
+
+    async fn gossip_contract_bytecode(
+        &self,
+        address: Address,
+        account_proof: &AccountProof,
+        block_hash: B256,
+        code_hash: B256,
+        code: Bytecode,
+    ) -> anyhow::Result<()> {
+        let permit = self
+            .gossip_limit_semaphore
+            .clone()
+            .acquire_owned()
+            .await
+            .expect("to be able to acquire semaphore");
+        let code_content_key = create_contract_content_key(address, code_hash)?;
+        let code_content_value = create_contract_content_value(block_hash, account_proof, code)?;
+        Self::spawn_serve_state_proof(
+            self.portal_client.clone(),
+            code_content_key,
+            code_content_value,
+            permit,
+            self.metrics.clone(),
+        );
+        Ok(())
+    }
+
+    async fn gossip_storage(
+        &self,
+        account_proof: &AccountProof,
+        storage_proof: &AccountProof,
```

Comment:
I think I commented on the other PR, but `AccountProof` should really be called `TrieProof` (or something similar) and it shows here.